piet-common/src/cairo_back.rs

```diff
@@ -101,41 +101,84 @@ impl<'a> BitmapTarget<'a> {
         CairoRenderContext::new(&mut self.cr)
     }
 
-    /// Get raw RGBA pixels from the bitmap.
-    pub fn into_raw_pixels(mut self, fmt: ImageFormat) -> Result<Vec<u8>, piet::Error> {
+    /// Get raw RGBA pixels from the bitmap by copying them into `buf`. If all the pixels were
+    /// copied, returns the number of bytes written. If `buf` wasn't big enough, returns an error
+    /// and doesn't write anything.
+    pub fn copy_raw_pixels(
```

Comment:
do these actually need `&mut self`? I guess if we need to flush the buffer or something?

Reply:
That's the current behavior, yes. Technically, cairo allows a flush without `&mut self`, but the d2d backend does require `&mut self`.